plugins/kubernetes/test/models/kubernetes/build_job_executor_test.rb

```diff
@@ -0,0 +1,152 @@
+        tag: 'latest', push: push, registry: registry_info)
+    end
+
+    let(:client) { stub }
+    let(:extension_client) { stub }
+
+    before do
+      Kubernetes::Cluster.any_instance.stubs(:client).returns client
+      Kubernetes::Cluster.any_instance.stubs(:extension_client).returns extension_client
+    end
+
+    it 'fails when an invalid registry is passed in' do
+      invalid_registry = registry_info.dup
+      invalid_registry[:serveraddress] = ''
+      extension_client.expects(:create_job).never
+      extension_client.expects(:delete_job).never
+      success, job_log = execute!(registry_info: invalid_registry)
+      refute success
+      assert_empty job_log
+    end
+
+    describe 'it can create a job' do
+      let(:build_job_config_path) { sample_file_path('kubernetes_job.yml') }
+      let(:job_config_file) { read_kubernetes_sample_file('kubernetes_role_config_file.yml') }
+      let(:job_container_spec) { stub }
+      let(:labels) { {foo: 'bar'} }
+      let(:build_config) do
+        stub(job: {
+          metadata: {
+            name: 'test_job',
+            namespace: 'test_ns',
+            labels: labels
+          },
+          spec: {
+            template: {
+              metadata: {
+                name: 'test_job',
+                namespace: 'test_ns',
+                labels: labels
+              },
+              spec: {
+                containers: [{ name: 'test-job' }]
+              }
+            }
+          }
+        })
+      end
+
+      before do
+        extension_client.expects(:delete_job).once
+        Kubernetes::RoleConfigFile.expects(:new).returns build_config
+        # build_config.job.stubs(:first_container).returns job_container_spec
+      end
+
+      around { |t| with_env "KUBE_BUILD_JOB_FILE": build_job_config_path, &t }
+
+      it 'reads job config file and populate correct job parameters, but fails to start a job' do
+        extension_client.expects(:create_job).raises(KubeException.new(500, 'Server Error', {}))
+
+        assert_raises KubeException do
+          execute!(registry_info: registry_info)
+        end
+
+        project_name_dash = project.permalink.tr('_', '-')
+        assert_match(/^#{Regexp.quote(project_name_dash)}-docker-build-#{build.id}-[0-9a-f]{14}$/,
+          build_config.job[:metadata][:name])
+
+        labels_config = [build_config.job[:metadata][:labels], build_config.job[:spec][:template][:metadata][:labels]]
+        labels_config.each do |v|
+          assert_equal v[:project], project_name_dash
+          assert_equal v[:role], 'docker-build-job'
+          assert_equal v[:foo], 'bar'
+        end
+
+        assert_equal build_config.job[:spec][:template][:spec][:containers][0][:args],
+          [project.repository_url, build.git_sha, project.docker_repo, 'latest', 'no', 'no']
+        assert_equal build_config.job[:spec][:template][:spec][:containers][0][:env].length, 1
+      end
+
+      describe 'can create job' do
```

Comment:
describes usually are something about a certain state like "describe 'when all params are good' do ... it 'creates a job'"